packages/datadog-plugin-pg/test/index.spec.js

```diff
@@ -685,7 +686,25 @@ describe('Plugin', () => {
 
             expect(queryText).to.equal(
               `/*dddb='postgres',dddbs='post',dde='tester',ddh='127.0.0.1',ddps='test',ddpv='${ddpv}',` +
-              `traceparent='00-${traceId}-${spanId}-00'*/ SELECT $1::text as message`)
+              `traceparent='00-${traceId}-${spanId}-01'*/ SELECT $1::text as message`)
+          })
+        })
+
+        it('query text should contain rejected sampling decision in the traceparent', async () => {
+          sinon.stub(tracer._tracer._prioritySampler, 'sample')
```

Comment:
Plugin tests are integration tests, mocking internals should be avoided whenever possible. Can the tracer be reconfigured with different rates instead?

Reply:
I tried at first but I couldn't manage to reconfigure the existing tracer instance with different sampling rates. My conclusion was that I would need to configure it on initialization, which would require a new test suite just to be able to a have a different tracer instance. 
So yes I agree with you, but I would like to see how.